userland/Configuration.mk

```diff
@@ -60,6 +60,9 @@ override CPPFLAGS += \
 	    -mpic-register=r9\
 	    -mno-pic-data-is-text-relative
 
+# Work around https://gcc.gnu.org/bugzilla/show_bug.cgi?id=85606
+override CPPFLAGS_cortex-m0 += -march=armv6s-m
```

Comment:
What was setting the `-march` flag in our compile system before? Is it normally not necessary?

Reply:
See the linked issue from Niklas, or the gcc bug linked in the comment right above. It's a gcc regression that requires explicitly stating the arch for m0 and m0+ cores.

That said, I need to grab a copy of that version of gcc to see why this didn't fix the problem, will try to do tomorrow

Reply:
I understand the issue that's causing this change. Before did we just not have to specify architecture at all? How was gcc figuring out which assembly instructions to use?

Reply:
it was/is figuring it out from the `-mcpu=cortex-m0` argument to gcc

Reply:
Oh, I see. That was enough before this regression for gcc to determine architecture. Maybe still specifying it is what's causing the issue to continue existing.

Reply:
Decent guess, but RIOT does the same thing as this, just appending the flag :/

@niklasad1 can you try something like:
`make && touch main.c && TOCK_ARCHS=cortex-m0 make V=1` for blink (should minimize the output to the part we care about), then maybe try futzing a bit just running the compile step that fails manually (try removing mcpu flag, try moving the flags around?). I don't really know what the needed fix is here